src/Service/Modules/Reports/PurchaseOrdersReportModule.cs

```diff
@@ -52,5 +67,32 @@ private async Task GetOrdersBySupplierReport(HttpRequest req, HttpResponse res)
 
             await res.Negotiate(results);
         }
+
+        private async Task GetOrdersBySupplierExport(HttpRequest req, HttpResponse res)
+        {
+            var resource = new OrdersBySupplierSearchResource();
+            resource.SupplierId = req.Query.As<int>("Id");
+            resource.From = req.Query.As<string>("FromDate");
+            resource.To = req.Query.As<string>("ToDate");
+
+            resource.Returns = req.Query.As<string>("Returns");
+            resource.Outstanding = req.Query.As<string>("Outstanding");
+            resource.Cancelled = req.Query.As<string>("Cancelled");
+            resource.Credits = req.Query.As<string>("Credits");
+            resource.StockControlled = req.Query.As<string>("StockControlled");
+
+            var results = this.purchaseOrderReportFacadeService.GetOrdersBySupplierExport(resource, req.HttpContext.GetPrivileges());
+
+            this.csvStreamWriter.WriteModel(results);
+
+            var contentDisposition = new ContentDisposition
+                                         {
+                                             FileName =
+                                                 $"ordersBySupplier{resource.From.Substring(0, 10)}_To_{resource.To.Substring(0, 10)}.csv"
+                                         };
+
+            this.stream.Position = 0;
```

Comment:
I think this could also be replaced with something like 
` res.Headers.Add("Content-Range", "bytes */*");`
Or whatever represents all bytes from what you were saying @lewisrenfrew, I don't have a preference